/** 
 * test solve dimension errors 
 */
@Test public void testSolveDimensionErrors(){
  final double[] refValues=new double[]{2.003,2.002,2.001,1.001,1.000,0.001};
  final RealMatrix matrix=EigenDecompositionTest.createTestMatrix(new Random(35992629946426l),refValues);
  DecompositionSolver es=new EigenDecomposition(matrix).getSolver();
  RealMatrix b=MatrixUtils.createRealMatrix(new double[2][2]);
  try {
    es.solve(b);
    Assert.fail("an exception should have been thrown");
  }
 catch (  MathIllegalArgumentException iae) {
  }
  try {
    es.solve(b.getColumnVector(0));
    Assert.fail("an exception should have been thrown");
  }
 catch (  MathIllegalArgumentException iae) {
  }
  try {
    es.solve(new ArrayRealVectorTest.RealVectorTestImpl(b.getColumn(0)));
    Assert.fail("an exception should have been thrown");
  }
 catch (  MathIllegalArgumentException iae) {
  }
}
